### PR TITLE
Better login/sign up error messages

### PR DESCRIPTION
**Pull request by k88hudson**
- Added a check if for connection via Android and new offline error messages
- Added more specific error handling for XHR errors, 400 errors, and other error errors
- Removed duplicate snack bar errors

I did not provide localized versions of the error messages in this patch, although I did move them to a an object on api. I'd prefer to do this as a follow up (filed #459) since there a number of ways we could do it, and i'd like to discuss what the best approach is

### CODE REVIEW

**Comment by k88hudson** [comment]
Filed https://github.com/mozilla/webmaker-core/issues/459

**Comment by k88hudson** [comment]
@gvn R?

File: src/pages/login/sign-in.jsx [hunk 1]
```diff
@@ -92,8 +91,9 @@ var SignIn = React.createClass({
         if (window.Platform) {
           window.Platform.trackEvent('Login', 'Sign In', 'Sign In Error');
         }
-        this.setState({globalError: true});
-        return reportError("Error while trying to log in", err);
+        this.setState({globalError: err.message || 'Something went wrong.'});
+        console.log(err);
```
Comment (gvn) [inline src/pages/login/sign-in.jsx:97]: Could do `console.error`

**Comment by gvn** [comment]
qvlpegi

**Pull request merged**